### PR TITLE
OP-593 Client findings

### PR DESCRIPTION
**Pull request by janjaap**
As per the client's request, this PR:
- allows legend labels to be hidden so that a graph can show only those labels that are important
- enables year cijfers to be filtered by specific years. The change can filter out odd years, even years, years before and years after a specific year. It can also filter out a specific set of years, whenever necessary
- adds a title to the `StackedBarChart` component
- adjusts the color coding logic
- reduces the number of labels along the x-axis of the `LineChart` component so that labels don't overlap
- corrects a couple of typos

### CODE REVIEW

File: src/services/util.js [hunk 1]
```diff
@@ -63,6 +64,7 @@ async function getConfigCijfers (gwb, config, recentOrAll = CIJFERS.ALL) {
       return {
         ...cijfers,
         label: c.label || cijfers.meta.label,
+        showInLegend: c.showInLegend !== undefined ? c.showInLegend : true,
```
Comment (Vegter) [inline src/services/util.js]: I prefer test for positive, reads easier. If x === y ? 1 : 2

File: src/services/util.js [hunk 2]
```diff
@@ -109,7 +119,21 @@ function getYearCijfers (data, last = null) {
         color: cijfer.color,
         display: (cijfer.waarde / totalWaarde[cijfer.jaar]) > 0.075 ? displayWaarde(cijfer) : '',
         cijfer
-      }))))
+      })))
+  ).filter(({ x }) => {
+    if (odd) return x % 2 === 0
+    if (even) return x % 2 > 0
```
Comment (Vegter) [inline src/services/util.js]: %2 is always 0 or 1

Comment (Vegter) [reply src/services/util.js]: Even = %2 === 0
Odd = %2 === 1

Comment (Vegter) [reply src/services/util.js]: Now it is just the other way araound

File: src/services/util.js [hunk 3]
```diff
@@ -109,7 +119,21 @@ function getYearCijfers (data, last = null) {
...
+    return true
+  }).filter(({ x }) => {
+    if (before) return x > before
```
Comment (Vegter) [inline src/services/util.js]: Looks like the comparison is just the other way around here also.
Before => x < before
Before = 2000, 1999 < 2000 is true
Or have I misunderstood something here?

File: src/services/util.js [hunk 4]
```diff
@@ -181,34 +205,49 @@ const displayWaarde = cijfer => {
...
+
+  configCijfers
+    .filter(({ showInLegend }) => showInLegend)
+    .forEach(({ label }) => {
+      if (!legendLabels.includes(label)) {
+        legendLabels.push(label)
+      }
+    })
+
+  return legendLabels
```
Comment (Vegter) [inline src/services/util.js]: You can use a set here and Array.from() to convert it back into an array.
The set is constructed from configCijfers.map(...)

File: test/unit/specs/util.spec.js [hunk 5]
```diff
@@ -230,6 +230,95 @@ describe('util', () => {
...
+    it('can get odd years for a given dataset', () => {
+      const yearCijfers = util.getYearCijfers(data, null, { even: true })
+      const years = getYears(yearCijfers)
+
+      expect(years).toEqual([2011, 2013, 2015])
+    })
+
+    it('can get even years for a given dataset', () => {
+      const yearCijfersEven = util.getYearCijfers(data, null, { odd: true })
+      const yearsEven = getYears(yearCijfersEven)
```
Comment (Vegter) [inline test/unit/specs/util.spec.js]: Seems strange, odd = true and that returns a list with even years

File: test/unit/specs/util.spec.js [hunk 6]
```diff
@@ -230,6 +230,95 @@ describe('util', () => {
...
+      // 'odd' takes precedence over 'even'
+      const yearCijfersEven2 = util.getYearCijfers(data, null, { odd: true, even: true })
+      const yearsEven2 = getYears(yearCijfersEven2)
+
+      expect(yearsEven2).toEqual([2012, 2014, 2016])
+    })
+
+    it('can get years before a given year for a given dataset', () => {
+      const yearCijfers = util.getYearCijfers(data, null, { after: 2014 })
+      const years = getYears(yearCijfers)
```
Comment (Vegter) [inline test/unit/specs/util.spec.js]: Same here. A better name should propably be skipAfter oid?

File: src/components/charts/HorizontalBarChart.vue [hunk 7]
```diff
@@ -63,6 +63,9 @@ export default {
         i
       }))
 
+      if (!vegaSpec.legends) vegaSpec.legends = [{}]
```
[the remark this answers is not in the document]
Comment (janjaap) [reply src/components/charts/HorizontalBarChart.vue]: I'll change all occurrences.

File: src/services/colorcoding.js [hunk 8]
```diff
@@ -135,7 +135,7 @@ function getCategory (zScore) {
   const average = 0.5
   const categories = [
     {
-      inCategory: s => s >= 2 * average,
+      inCategory: s => 2 * average <= s,
```
[the remark this answers is not in the document]
Comment (janjaap) [reply src/services/colorcoding.js]: I changed the notation so that it is in line with the calculation provided by the client.

**Pull request merged**